File: GuestPass.Api/models/Event.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GuestPass.Api.Models;

[Table("events")]
public class Event
{
    [Key]
    [Column("id")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("location")]
    public string Location { get; set; } = string.Empty;

    [Column("date")]
    public DateTimeOffset Date { get; set; }

    [Column("ownerid")]
    public Guid? CreatedBy { get; set; }

    [Column("createdat")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public DateTimeOffset CreatedAt { get; set; }

    // Navigation properties
    [ForeignKey("CreatedBy")]
    public Profile? Owner { get; set; }

    public ICollection<Guest> Guests { get; set; } = new List<Guest>();
}

File: guestpass-mobile/lib/data/models/event.dart
/// Model Event
class Event {
  final String id;
  final String name;
  final String location;
  final DateTime date;
  final String? createdBy;
  final DateTime createdAt;

  Event({
    required this.id,
    required this.name,
    required this.location,
    required this.date,
    this.createdBy,
    required this.createdAt,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] as String,
      name: json['name'] as String,
      location: json['location'] as String,
      date: DateTime.parse(json['date'] as String),
      createdBy: json['createdBy'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
